distributed/deploy/old_ssh.py

```diff
@@ -349,12 +350,27 @@ def __init__(
         nanny_port=None,
         remote_dask_worker="distributed.cli.dask_worker",
         local_directory=None,
+        **kwargs,
```

Comment:
Adding `**kwargs` here will cause invalid keyword arguments to go unnoticed. For example, today a typo of `memory_limt` instead of `memory_limit` will raise an error, while with this change as is such a typo will be silently ignored.

Instead I'd rather us either keep an explicit `nprocs=` keyword around, but still raise a `ValueError` / `FutureWarning` accordingly, or do something like (below is pseudocode)

```python
nprocs = kwargs.pop("nprocs")
if kwargs:
    raise TypeError("...")
<existing ValueError / FutureWarning logic>